src/Common/src/Interop/Unix/libcurl/Interop.SafeCurlHandle.cs

```diff
@@ -37,6 +40,19 @@ protected override bool ReleaseHandle()
 
         internal sealed class SafeCurlMultiHandle : SafeHandle
         {
+            private readonly CancellationTokenSource _cancellationTokenSource = new CancellationTokenSource();
```

Comment:
Sorry, I didn't understand the response.  What does CancellationTokenSource on the handle buy you over just having a volatile Boolean on the handle?  If there's a good reason for it, then it's fine.  But typically you use a CTS because either a) you need an object you can hand around and/or b) you need to be able to Register for callbacks.  Here, you're not doing (b), and you already have (a) (the SafeCurlMultiHandle), so I'm not sure what value the CTS provides in this case.